test/batch.js

```diff
@@ -0,0 +1,80 @@
+const { expect } = require('chai')
+const property   = require('prop-factory')
+
+const { batch, evolveP } = require('..')
+
+describe('batch', () => {
+  const res = property()
+
+  const obj = { a: 'a', b: 'b', c: 'c' }
+
+  describe('with no options', () => {
+    const asyncListFn = ([a, b, c]) =>
+      Promise.resolve([a, b, c])
+
+    const batched = batch(undefined, asyncListFn)
+
+    const test = evolveP({
+      a: batched,
+      b: batched,
+      c: batched
+    })
+
+    beforeEach(() =>
+      test(obj).then(res)
+    )
+
+    it('has safe defaults', () =>
+      expect(res()).to.eql(obj)
```

Comment:
Does it need something like `expect(asyncListFn).to.have.been.calledOnce`?

Reply:
Can do.